src/main.rs

```diff
@@ -133,88 +132,114 @@ async fn main() {
+                    network_name,
+                    provider_name
+                );
+                continue;
+            };
+
             let block_number =
                 U64::as_u64(&indexing_network_provider.get_block_number().await.unwrap()) - 5;
+
+            let block_clock = block_store
+                .entry(network_name)
+                .or_insert_with(|| BlockClock {
+                    current_block: 0,
+                    compare_block: 0,
+                });
+
+            if block_clock.current_block == block_number {
+                sleep(Duration::from_secs(5));
+                continue;
+            }
+
+            debug!("{} {}", "🔗 Block number:".cyan(), block_number);
+            block_clock.current_block = block_number;
+
+            if block_number == block_clock.compare_block {
+                debug!("{}", "Comparing attestations".magenta());
+
+                let remote_attestations = process_messages(
+                    Arc::clone(MESSAGES.get().unwrap()),
+                    &registry_subgraph,
+                    &network_subgraph,
+                )
+                .await;
+                match remote_attestations {
+                    Ok(remote_attestations) => {
+                        let mut messages = MESSAGES.get().unwrap().lock().unwrap();
+                        match compare_attestations(
+                            block_clock.compare_block - wait_block_duration,
+                            remote_attestations,
+                            Arc::clone(&local_attestations),
+                        ) {
+                            Ok(msg) => {
+                                debug!("{}", msg.green().bold());
+                                messages.clear();
+                            }
+                            Err(err) => {
+                                error!("{}", err);
+                                messages.clear();
+                            }
+                        }
+                    }
+                    Err(err) => {
+                        error!(
+                            "{}{}",
+                            "An error occured while parsing messages: {}".red().bold(),
+                            err
+                        );
+                    }
+                }
+            }
+
+            let poi_query =
+                partial!( query_graph_node_poi => graph_node_endpoint.clone(), id.clone(), _, _);
+
             // Send POI message at a fixed frequency
-            let examination_frequency = SUPPORTED_NETWORK_INTERVALS.get(&network[..]).expect("Subgraph is indexing an unsupported network, please report an issue on https://github.com/graphops/graphcast-rs");
+            let examination_frequency = NETWORKS
+                .iter()
+                .find(|n| n.name.to_string() == network_name.to_string())
+                .map(|n| n.interval)
+                .unwrap_or_else(|| {
+                    panic!(
```

Comment:
Maybe we should just use `warn!` so that the rest of networks can still be compared

Reply:
oh yeah, I thought I already removed all the panics :derp: